Firestore/core/src/firebase/firestore/remote/serializer.cc

```diff
@@ -141,76 +147,80 @@ std::string DecodeString(pb_istream_t* stream) {
   return result;
 }
 
-}  // namespace
-
-using firebase::firestore::model::FieldValue;
-
-void Serializer::EncodeFieldValue(const FieldValue& field_value,
-                                  std::vector<uint8_t>* out_bytes) {
-  // TODO(rsgowman): how large should the output buffer be? Do some
-  // investigation to see if we can get nanopb to tell us how much space it's
-  // going to need.
-  uint8_t buf[1024];
-  pb_ostream_t stream = pb_ostream_from_buffer(buf, sizeof(buf));
+void EncodeObject(
+    pb_ostream_t* stream,
+    const std::map<const std::string, const FieldValue>& object_value);
+std::map<const std::string, const FieldValue> DecodeObject(
+    pb_istream_t* stream);
 
+// Named '..Impl' so as to not conflict with Serializer::EncodeFieldValue.
```

Comment:
Ah, sorry, I missed your comment with the explanation. So one shadows the other. It's also possible to use the free function as `::EncodeFieldValue`, but I guess this code will change quite a bit after refactoring anyway.

Reply:
That doesn't work either. (It was actually the first thing I though to try.) I presume because it's not actually in the global namespace. And yeah, it should change significantly after refactoring.

Reply:
Yes, the declaration in the class shadows the declaration in the containing namespace, so the free function doesn't even participate in the overload resolution. I'm pretty sure that `::firebase::firestore::whatever::NameOfFunction` should work (sorry I communicated it poorly), but if it doesn't, guess I'm wrong.